pages/dao/[symbol]/proposal/components/instructions/CreateAssociatedTokenAccount.tsx

```diff
@@ -0,0 +1,174 @@
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import useWalletStore from 'stores/useWalletStore'
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import Select from '@components/inputs/Select'
+import {
+  ProgramAccount,
+  serializeInstructionToBase64,
+  Governance,
+} from '@solana/spl-governance'
+import GovernedAccountSelect from '../GovernedAccountSelect'
+import { createAssociatedTokenAccount } from '@utils/associated'
+import { getSplTokenMintAddressByUIName, SPL_TOKENS } from '@utils/splTokens'
+
+const CreateAssociatedTokenAccount = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const { realmInfo } = useRealm()
+  const [governedAccounts, setGovernedAccounts] = useState<
+    GovernedMultiTypeAccount[]
+  >([])
+  const { getGovernedMultiTypeAccounts } = useGovernanceAssets()
+
+  useEffect(() => {
+    getGovernedMultiTypeAccounts().then(setGovernedAccounts)
+  }, [])
+
+  const shouldBeGoverned = index !== 0 && governance
+  const programId: PublicKey | undefined = realmInfo?.programId
+  const [form, setForm] = useState<CreateAssociatedTokenAccountForm>({})
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+
+    if (
+      !connection ||
+      !isValid ||
+      !programId ||
+      !form.governedAccount?.governance?.account ||
+      !form.splTokenMintUIName ||
+      !wallet?.publicKey
+    ) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: form.governedAccount?.governance,
+      }
+    }
+
+    const [tx] = await createAssociatedTokenAccount(
+      // fundingAddress
+      wallet.publicKey,
+
+      // walletAddress
+      form.governedAccount.governance.pubkey,
+
+      // splTokenMintAddress
+      getSplTokenMintAddressByUIName(form.splTokenMintUIName)
+    )
+
+    console.log('infos', {
+      governanceAccount: form.governedAccount.governance.account,
+      governanceOwner: form.governedAccount.governance.owner.toString(),
+      fundingAddress: wallet.publicKey.toString(),
+      walletAddress: form.governedAccount.governance.pubkey.toString(),
+      splTokenMintName: form.splTokenMintUIName,
+      splTokenMintAddress: getSplTokenMintAddressByUIName(
+        form.splTokenMintUIName
+      ).toString(),
+    })
+
+    console.log('tx', tx)
```

Comment:
remove if not strictly needed, if strictly needed then make logs descriptive, easy to read, non ambiguous

Reply:
with the latest fix on governance select, this cast doesn't exist anymore